File: hw/bonilla/u3/hw33StrategyCode/Strategy/src/main/java/ec/edu/espe/strategy/model/SortingContext.java
package ec.edu.espe.strategy.model;

/**
 *
 * @author Arelis Samantha Bonilla Cruz, Student, @ESPE
 */
public class SortingContext {

    private SortingStrategy ss;

    public int[] sort(int[] data) {
        int size = data.length;
        ss = setSortStrategy(size);
        System.out.println("Selected strategy: " + ss.getClass().getSimpleName());
        return ss.sort(data);
    }

    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) {
            ss = new BubbleSort();
        } else if (n >= 30 && n < 100) {
            ss = new InsertionSort();
        } else {
            ss = new QuickSort();
        }
        return ss;
    }
}

File: exam/vargas/u3/q47_100/javaScript/model/SortingContext.js
const BubbleSort = require("./BubbleSort");
const InsertionSort = require("./InsertionSort");
const QuickSort = require("./QuickSort");

class SortingContext {
  constructor() {
    this.strategy = null;
  }

  sort(data) {
    const n = data.length;
    this.strategy = this.setStrategy(n);
    return this.strategy.sort(data);
  }

  setStrategy(n) {
    if (n > 0 && n < 30) return new BubbleSort();
    if (n >= 30 && n < 100) return new InsertionSort();
    return new QuickSort();
  }

  getStrategyName() {
    return this.strategy.constructor.name;
  }
}

module.exports = SortingContext;
